File: 2024/452. Minimum Number of Arrows to Burst Balloons.py
class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        def sortFunc(interval: List[int]):
            return (interval[1], interval[0])
        points.sort(key=sortFunc)
        count = 0
        i = 0
        while i < len(points):
            currEnd = points[i][1]
            count += 1
            while i < len(points) and currEnd >= points[i][0]:
                i += 1
        return count

File: 2024/452. Minimum Number of Arrows to Burst Balloons.cpp
class Solution {
public:
    int findMinArrowShots(vector<vector<int>>& points) {
        sort(points.begin(), points.end(), [](vector<int>& left, vector<int>& right) {
            return left[1] == right[1] ? left[0] < right[0] : left[1] < right[1];
        });
        int count = 1, prev = points[0][1];
        for (int i = 1; i < points.size(); i++) {
            if (points[i][0] <= prev) {
                prev = min(points[i][1], prev);
            } else {
                prev = points[i][1];
                count += 1;
            }
        }
        return count;
    }
};